src/Workleap.DomainEventPropagation.Publishing/Extensions/ServiceCollectionExtensions.cs

```diff
@@ -36,19 +37,24 @@ public static IEventPropagationPublisherBuilder AddEventPropagationPublisher(thi
             using var sp = services.BuildServiceProvider();
 
             var options = sp.GetRequiredService<IOptions<EventPropagationPublisherOptions>>().Value;
-
             var topicEndpointUri = new Uri(options.TopicEndpoint);
-            var topicCredentials = new AzureKeyCredential(options.TopicAccessKey);
-
-            builder
-                .AddEventGridPublisherClient(topicEndpointUri, topicCredentials)
-                .WithName(EventPropagationPublisherOptions.ClientName)
-                .ConfigureOptions(clientOptions =>
-                {
-                    clientOptions.Retry.Mode = RetryMode.Fixed;
-                    clientOptions.Retry.MaxRetries = 1;
-                    clientOptions.Retry.NetworkTimeout = TimeSpan.FromSeconds(4);
-                });
+
+            if (options.TokenCredential is not null)
```

Comment:
How do we expect people to configure this TokenCredential? Can't really be done in appsettings.json I think, so it would have to be through code. Will probably be good to make sure it's in the readme